Mood details

adityadipakpatel opened the pull request:
Resolves: #110 #118 #127 

 - Add Glide dependency for image support
 - Add new parameters to CreateMoodEvent, MoodEvent.java
 - Mood Details page

 - What Did you change?
     - Implemented the Mood Details Page
     - Implemented Navigation from Home Feed to View Details Page
 - How should a reviewer test?
     - Run the MoodDetailsNavigationTest.java

**Reviewer Checklist**

- [x] Are all relevant issues referenced by this PR?
- [x] Does the app still build?
- [x] Do all tests pass?
- [x] Does the feature work as expected?
- [x] Does the feature still keep the app bug free?
- [x] Is the description of the changes correct/present?

adityadipakpatel committed:
Add new parameters in MoodEvent.java

 - add getters, setters for parameters like trigger, social situation, reason, photo, location(latitude, longitude)
 - not sure how to handle photo storage

adityadipakpatel committed:
Dependencies Updates

adityadipakpatel committed:
Update navigation graph

 - for Mood Details
 - for Edit Mood, because Edit Mood has not been started yet

adityadipakpatel committed:
Mood Details, Edit Mood Files

 - UI xml files will be updated later on(UI is not ready yet)

adityadipakpatel committed:
Setup MoodClickListener in Home Feed

 - to dynamically go back to previos screen(home feed/display filtered moods) not to home feed when user clicks the "back" button

adityadipakpatel committed:
Update MoodAdapter for MoodClickListener

adityadipakpatel committed:
update database collection name

 - because there wont be just one mood event, there will be multiple mood events in the collection/database

adityadipakpatel committed:
Add parameters to CreateMoodEvent

 - added trigger, social situation, reason, photo, location(latitude, longitude)

adityadipakpatel committed:
Add Glide dependency for image support

 - https://github.com/bumptech/glide

adityadipakpatel committed:
Merge branch 'main' of github.com:cmput301-w25/project-kernelcrew into mood_details

adityadipakpatel committed:
revert db name to how it was

adityadipakpatel committed:
updates to test files

adityadipakpatel committed:
updates to UI

adityadipakpatel committed:
in google material UI now

adityadipakpatel committed:
some UI changes

Commit:
Add .gitignore

adityadipakpatel committed:
Fix some bugs

adityadipakpatel committed:
UI colors a little better

adityadipakpatel committed:
Merge branch 'main' of github.com:cmput301-w25/project-kernelcrew into mood_details

adityadipakpatel committed:
some ui color updates, 1 mood/element still not in 1 box

adityadipakpatel committed:
UI updates

adityadipakpatel committed:
change home feed test back to how it was

adityadipakpatel committed:
some testcase update(still failing)

adityadipakpatel committed:
version upgrade

adityadipakpatel committed:
back to old nav graph

adityadipakpatel committed:
updated nav graph for (home feed to mood details), (mood details to edit mood)

adityadipakpatel committed:
navigation from home feed to mood details

adityadipakpatel committed:
updates to tests

adityadipakpatel committed:
db name bug fixes

adityadipakpatel committed:
missing imports

adityadipakpatel committed:
bump dependencies

adityadipakpatel committed:
HomeFeedNavigationTest to how it was

adityadipakpatel committed:
dependencies changes

adityadipakpatel committed:
some updates to revert some bad changes

adityadipakpatel committed:
gradle files updates

 - home feed test now works
 - mood details test still fails

PossiblyAShrub commented:
From CI:
>   /home/runner/work/project-kernelcrew/project-kernelcrew/code/app/src/main/AndroidManifest.xml: Error: When targeting Android 13 or higher, posting a permission requires holding the POST_NOTIFICATIONS permission (usage from com.bumptech.glide.request.target.NotificationTarget) [NotificationPermission]
>  
>     Explanation for issues of type "NotificationPermission":
>     When targeting Android 13 and higher, posting permissions requires holding
>     the runtime permission android.permission.POST_NOTIFICATIONS.

Looks like it has been a known issue for a while... I don't think we need this permission. We can ignore this for now, I can find a way to suppress it.

adityadipakpatel committed:
some dependencies updates

adityadipakpatel committed:
add authetication

adityadipakpatel committed:
this test now works

adityadipakpatel committed:
some changes similar to home feed test

adityadipakpatel committed:
bug still exists

 - moodEventId does exist but documentSnapshot does not exist

adityadipakpatel committed:
bug fix in Emotion.java

adityadipakpatel committed:
remove "TODO" line

adityadipakpatel committed:
some updates, remove log lines

adityadipakpatel committed:
all tests pass for MoodDetailsNavigationTest

 - line 137, 138

adityadipakpatel committed:
a small change for lint error

adityadipakpatel committed:
change for lint

adityadipakpatel committed:
remove excess sleep

adityadipakpatel committed:
Merge branch 'main' into mood_details

adityadipakpatel committed:
dependencies updates, merge conflicts fix

adityadipakpatel committed:
update MyProfileTest.java

 - added seedDatabase in @BeforeClass
 - added @After

adityadipakpatel committed:
remove duplicate line

adityadipakpatel committed:
increase sleep because android.yml is failing

adityadipakpatel committed:
updates to Mood Details Test

 - viewDetails button is shown in home feed but is a part of item_mood(child), so changes for clicking a button in child element
